src/components/VideoProvider/useBackgroundSettings/useBackgroundSettings.ts

```diff
@@ -65,13 +68,61 @@ export const backgroundConfig = {
   images,
 };
 
-// TODO : Add video processing logic after backgroundSettings change
-// useEffect hooks, etc ...
+const virtualBackgroundAssets = '/virtualbackground';
 
-export default function useBackgroundSettings() {
-  const [backgroundSettings, setBackgroundSettings] = useState({
-    type: 'none',
-    index: 0,
-  } as BackgroundSettings);
-  return [backgroundSettings, setBackgroundSettings] as const;
+let blurProcessor: GaussianBlurBackgroundProcessor;
+
+export default function useBackgroundSettings(videoTrack: LocalVideoTrack) {
+  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(() => {
+    const localStorageSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
+    return localStorageSettings ? JSON.parse(localStorageSettings) : { type: 'none', index: 0 };
+  });
+
+  // load processors on initialization
+  useEffect(() => {
+    if (!blurProcessor) {
+      blurProcessor = new GaussianBlurBackgroundProcessor({
+        assetsPath: virtualBackgroundAssets,
+      });
+      blurProcessor.loadModel();
+    }
+  }, []);
+
+  // for reapplying background replacement after camera is toggled on again
+  useEffect(() => {
+    const storedBackgroundSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
+    if (videoTrack && storedBackgroundSettings) {
+      const parsedSettings = JSON.parse(storedBackgroundSettings!);
+      if (parsedSettings.type === 'blur') {
+        videoTrack.addProcessor(blurProcessor);
```

Comment:
This seems duplicative of what is already in `updateBackgroundSettings`. Also, I don't think we need to read from local storage every time the user enables their video. We can just rely on the `backgroundSettings` variable, which gets its initial value from local storage. 

After thinking about this hook some, it seems like we just need to run the logic in `updateBackgroundSettings` whenever the settings are updated, or whenever the video track changes. In my mind this calls for a useEffect hook. The only state here is the selected background settings, and the video track. The application of the processor, and saving the settings to local storage are just side-effects of the changes in state. So I think all the code inside of `updateBackgroundSettings` can go in a useEffect hook instead, which will be set up to react to any changes in `backgroundSettings` or `videoTrack`. Instead of passing `updateBackgroundSettings` to the videoProvider, we could just pass `setBackgroundSettings` instead.

What do you think about a hook like this (it incorporates changes from my other comments):

```
let blurProcessor = new GaussianBlurBackgroundProcessor({
  assetsPath: '/virtualbackground',
});
blurProcessor.loadModel();

export default function useBackgroundSettings(videoTrack: LocalVideoTrack) {
  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(() => {
    const localStorageSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
    return localStorageSettings ? JSON.parse(localStorageSettings) : { type: 'none', index: 0 };
  });

  useEffect(() => {
    if (videoTrack) {
      if (videoTrack.processor) {
        videoTrack.removeProcessor(videoTrack.processor);
      }
      if (backgroundSettings.type === 'blur') {
        videoTrack.addProcessor(blurProcessor);
      } else if (backgroundSettings.type === 'image') {
        // TODO add image replacement logic
      }
    }
    window.localStorage.setItem(BG_SETTINGS_KEY, JSON.stringify(backgroundSettings));
  }, [videoTrack, backgroundSettings]);

  return [backgroundSettings, setBackgroundSettings] as const;
}
```

Reply:
Hi Tim, after thinking about how to simplify this hook, I managed to simplify its implementation down to something similar to this! After some testing, this also takes care of an important edge case (reapplying the previously selected background settings after a user turns off then turns on their camera)